brush-core/src/extendedtests.rs

```diff
@@ -182,7 +182,10 @@ pub(crate) fn apply_unary_predicate_to_str(
         }
         ast::UnaryPredicate::ShellVariableIsSetAndAssigned => Ok(shell.env.is_set(operand)),
         ast::UnaryPredicate::ShellVariableIsSetAndNameRef => {
-            error::unimp("unary extended test predicate: ShellVariableIsSetAndNameRef")
+            match (shell.env.is_set(operand), shell.env.get(operand)) {
```

Comment:
This should fix the behavior, but it may incur 2 environmental lookups (which might require going through multiple scopes to find the match).

It would be cleaner if we could sort this out with just a single call to `shell.env.get()`. Underneath, the `is_set()` function just checks if the variable exists and doesn't match `ShellValue::Unset(_)`. Either we could do that inline here, or if we don't already have an `is_set()` helper on `ShellValue`, we could just add one. It's a common enough thing that we need to do.

Reply:
> This should fix the behavior, but it may incur 2 environmental lookups (which might require going through multiple scopes to find the match).
> 
> It would be cleaner if we could sort this out with just a single call to `shell.env.get()`. Underneath, the `is_set()` function just checks if the variable exists and doesn't match `ShellValue::Unset(_)`. Either we could do that inline here, or if we don't already have an `is_set()` helper on `ShellValue`, we could just add one. It's a common enough thing that we need to do.

https://github.com/reubeno/brush/pull/439#discussion_r2056680824

I added the `is_set` because I thought you said it had to explicitly be set?

Reply:
To clarify, the change looks functionally correct. My last comment was focused on avoiding a double lookup. 

The `is_set()` call's implementation does an env lookup + check on the retrieved value (if the `ShellValue` object exists), then discards the retrieved value. The `get()` call then performs a second lookup.

My proposal is basically to do exactly what `is_set()` does internally (i.e. still check for existing-but-unset case) but not discard the retrieved value.

Is that clearer?